### PR TITLE
Version Checker with Branching for Checks on provideDiagnosisKeys below v16 of ENF (EXPOSUREAPP-2213)

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
This PR features a small class for checking the API Version to branch off logic. It also includes a branching for provideDiagnosisKeys in the RetrieveDiagnosisKeysTransaction. It can be tested by comparing devices with different versions of GMS and seeing how many provideDiagnosisKeys calls are executed in the transaction.

Expectations:
- Versions always have exactly 8 digits, and can be compared by using their raw value
Cases:
- `<16xxxxxx` should query with 14 calls
- `<15xxxxxx` an exception is converted to 14 calls as we expect getVersion() to be unavailable due to a too old version of ENF
- `>=16xxxxxx` should query with 1 call

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/GoogleAPIVersion.kt
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/GoogleAPIVersion.kt:8]: Maybe make this injectable. `@Inject constructor()`. The class is stateless, but can be recreated, so we can also tag the class `@Reusable`.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/GoogleAPIVersionTest.kt
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/GoogleAPIVersionTest.kt:30]: You could annotate the whole class with that instead of each function.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/GoogleAPIVersionTest.kt
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/GoogleAPIVersionTest.kt:75]: Personally I prefer AfterEach right below BeforeEach to keep a better overview of what is being setup and torn down. what do you think?

**Review by d4rken — APPROVED** [review]
lgtm, waiting for CircleCI after dev merge

**Review by BMItr — APPROVED** [review]

**Comment by vaubaehn** [comment]
This is even better than #1088...
According to the label, is there any chance to find this in the release build of 1.4.0? That would be awsome!

Just one minor suggestion (hope not to bother...): The name of method `isAbove` implies a GreaterThan-logic for me, while the intended logic is GreaterOrEqual. Maybe re-naming to `isAtLeast` to avoid later confusion?

**Comment by d4rken** [comment]
> According to the label, is there any chance to find this in the release build of 1.4.0? That would be awesome!

Likely :slightly_smiling_face: 

> Just one minor suggestion (hope not to bother...): The name of method isAbove implies a GreaterThan-logic for me, while the intended logic is GreaterOrEqual. Maybe re-naming to isAtLeast to avoid later confusion?

Good idea :+1:  #1159

**Pull request merged**
